main.py

```diff
@@ -34,15 +51,8 @@ def plot_bar_graph(x_data, y_data, x_label, y_label, title):
         break;
 
     if (user_input >= 1) and (user_input <= len(states)):
-        cursor.execute(sql.SQL("select year from public.amazon"))
-        years = pd.DataFrame(cursor.fetchall()).drop_duplicates().iloc[:, 0]
-        query = "SELECT {} FROM public.amazon WHERE state = %s;"
-        columns = ("year", "number")
-        state = [states[user_input - 1]]
-        cursor.execute(sql.SQL(query).format(sql.SQL(', ').join(map(sql.Identifier, columns))), state)
-        data_acquired = pd.DataFrame(data=cursor.fetchall(), columns=columns)
-        print(type(years))
-        print(type(data_acquired['year']))
+        data_acquired = data = get_plot_data(amazon,engine,states[user_input])
```

Comment:
Good catch here.
Removing all logic from a block of code and separate it on different functions. (`Separation of concerns` is a design pattern)
Way to go. 🚀

Reply:
Thank you for the positive reinforcement :D